labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -45,7 +52,10 @@ object RecursionPatternMatchingExercise {
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
   def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case Nil => Nil
+      case a :: _ => (in filter (_ == a)) :: groupEquals(in filterNot (_ == a))
```

Comment:
This needs to travers the list twice in each call to the function: once for `filter` and once for `filterNot`. A more efficient answer would use `partition`.